packages/react-native-editor/__device-tests__/helpers/utils.js

```diff
@@ -203,49 +203,22 @@ const stopDriver = async ( driver ) => {
 	}
 };
 
-/*
- * Problems about the 'clear' parameter:
- *
- * On Android: "clear" is defaulted to true because not clearing the text requires Android to use ADB, which
- * has demonstrated itself to be very flaky, particularly on CI. In other words, clear the view unless you absolutely
- * have to append the new text and, in that case, append fewest number of characters possible.
- *
- * On iOS: "clear" is not defaulted to true because calling element.clear when a text is present takes a very long time (approx. 23 seconds)
- */
 const typeString = async ( driver, element, str, clear ) => {
 	if ( isKeycode( str ) ) {
-		return await pressKeycode( driver, getKeycode( str ) );
+		const keyCode = isAndroid() ? getKeycode( str ) : str;
+		return await pressKeycode( driver, keyCode );
 	}
-	if ( isAndroid() ) {
-		await typeStringAndroid( driver, element, str, clear );
-	} else {
-		await typeStringIos( driver, element, str, clear );
-	}
-};
 
-const typeStringIos = async ( driver, element, str, clear ) => {
 	if ( clear ) {
-		// await element.clear(); This was not working correctly on iOS so need a custom implementation
-		await clearTextBox( driver, element );
+		await element.clearValue();
 	}
-	await element.type( str );
 
-	// Wait for the list auto-scroll animation to finish
-	await driver.sleep( 3000 );
-};
+	await element.addValue( str );
 
-const clearTextBox = async ( driver, element ) => {
-	await element.click();
-	let originalText = await element.text();
-	let text = originalText;
-	// We are double tapping on the text field and pressing backspace until all content is removed.
-	do {
-		originalText = await element.text();
-		await doubleTap( driver, element );
-		await element.type( '\b' );
-		text = await element.text();
-		// We compare with the original content and not empty because text always return any hint set on the element.
-	} while ( originalText !== text );
+	if ( ! isAndroid() ) {
+		// Wait for the list auto-scroll animation to finish
```

Comment:
What is the "list" in this context of typing a string? Will you expand this comment to better describe what is occurring here please?

Reply:
Of course! I think it'd be a good idea to expand on this inline comment and provide more context.

When we refactored the KeyboardAwareFlatList, we introduced [useScrollToTextInput](https://github.com/WordPress/gutenberg/blob/9d15719d3d1b7d43058398001d542712d9b42739/packages/components/src/mobile/keyboard-aware-flat-list/use-scroll-to-text-input.native.js) which scrolls the main block list depending on the caret position in the screen.

This directly affects visual tests when interacting with the scroll, you'd get flaky tests due to sometimes capturing the screenshot before the scroll has finished moving after typing.

I've also seen that other typing-related tests improved stability after adding this delay.

Reply:
Ah, thank you for explaining. I suggest expanding the inline comment with this nuance. 

```suggestion
		// Await the completion of the scroll-to-text-input animation
```

Reply:
Cool! Sounds good to me, thanks!